cli/cmd/install.go

```diff
@@ -135,8 +227,86 @@ func validateAndBuildConfig(options *installOptions) (*installConfig, error) {
 		options.proxyMemoryRequest = "20Mi"
 	}
 
+	var identity *installIdentityConfig
+	if idopts := options.identityOptions; idopts != nil {
+		trustDomain := idopts.trustDomain
+		if trustDomain == "" {
+			return nil, errors.New("Trust domain must be specified")
+		}
+		issuerName := fmt.Sprintf("identity.%s.%s", controlPlaneNamespace, trustDomain)
+
+		// Load signing material from options...
+		if idopts.trustPEMFile != "" || idopts.crtPEMFile != "" || idopts.keyPEMFile != "" {
+			if idopts.trustPEMFile == "" {
+				return nil, errors.New("a trust anchors file must be specified if other credentials are provided")
+			}
+			if idopts.crtPEMFile == "" {
+				return nil, errors.New("a certificate file must be specified if other credentials are provided")
+			}
+			if idopts.keyPEMFile == "" {
+				return nil, errors.New("a private key file must be specified if other credentials are provided")
+			}
+
+			// Validate credentials...
+			creds, err := tls.ReadPEMCreds(idopts.keyPEMFile, idopts.crtPEMFile)
+			if err != nil {
+				return nil, err
+			}
+
+			trustb, err := ioutil.ReadFile(idopts.trustPEMFile)
+			if err != nil {
+				return nil, err
+			}
+			trustAnchorsPEM := string(trustb)
+			roots, err := tls.DecodePEMCertPool(trustAnchorsPEM)
+			if err != nil {
+				return nil, err
+			}
+
+			issuerName := "" // TODO restrict issuer name?
```

Comment:
By _restrict issuer name_, are you thinking about eventually support whitelisted issuers?

Reply:
well, when we generate an issuer, we give it the name `identity.NS.TRUST-DOMAIN...` -- we could potentially enforce that naming for issuers, but it's not high-priority.